r2dbc/boot-reactive-cache/src/main/java/com/example/cache/services/MovieService.java

```diff
@@ -19,55 +22,67 @@ public class MovieService {
     private final MovieRepository movieRepository;
     private final MovieMapper movieMapper;
     private final ReactiveRedisTemplate<String, Movie> reactiveRedisTemplate;
+    private final ReactiveValueOperations<String, Movie> stringMovieReactiveValueOperations;
 
+    @Autowired
     public MovieService(
             MovieRepository movieRepository,
             MovieMapper movieMapper,
             ReactiveRedisTemplate<String, Movie> reactiveRedisTemplate) {
         this.movieRepository = movieRepository;
         this.movieMapper = movieMapper;
         this.reactiveRedisTemplate = reactiveRedisTemplate;
+        this.stringMovieReactiveValueOperations = reactiveRedisTemplate.opsForValue();
     }
 
     public Flux<MovieResponse> findAll() {
         return reactiveRedisTemplate
-                .keys("movie:*")
+                .keys(AppConstants.MOVIE_KEY + "*")
                 // Fetching cached movies.
-                .flatMap(key -> reactiveRedisTemplate.opsForValue().get(key))
+                .flatMap(stringMovieReactiveValueOperations::get)
                 // If cache is empty, fetch the database for movies
                 .switchIfEmpty(movieRepository
                         .findAll()
                         // Persisting the fetched movies in the cache.
-                        .flatMap(movie ->
-                                reactiveRedisTemplate.opsForValue().set(AppConstants.MOVIE_KEY + movie.id(), movie))
-                        // Fetching the movies from the updated cache.
-                        .thenMany(reactiveRedisTemplate
-                                .keys(AppConstants.MOVIE_KEY + "*")
-                                .flatMap(key ->
-                                        reactiveRedisTemplate.opsForValue().get(key))))
+                        .flatMap(
```

Comment:
@coderabbitai, how to update the cache asynchronously? here I am updating the cache and then responding to the caller

Reply:
@coderabbitai, I want to achieve for `findAll` method you have provided for `saveAllMovies`

Reply:
@coderabbitai, 

The below suggested code 

```java
public Flux<MovieResponse> findAll() {
    return reactiveRedisTemplate
            .keys(AppConstants.MOVIE_KEY + "*")
            .flatMap(stringMovieReactiveValueOperations::get)
            .switchIfEmpty(movieRepository
                    .findAll()
                    .doOnNext(movie -> 
                        Mono.defer(() -> stringMovieReactiveValueOperations.set(AppConstants.MOVIE_KEY + movie.id(), movie))
                            .subscribeOn(Schedulers.boundedElastic()) // Perform cache update asynchronously
                            .subscribe() // Asynchronously update the cache
                    ))
            .map(movieMapper::toResponse);
}
```  

is not returning the movie value, I want to update cache Asynchronously and return the movie object as well

Reply:
@coderabbitai,

Without disturbing the method signature, how to achieve above for below code 

```java
 public Mono<MovieResponse> findMovieById(Long id) {
        return reactiveRedisTemplate
                .opsForValue()
                .get(AppConstants.MOVIE_KEY + id)
                .switchIfEmpty(movieRepository
                        .findById(id)
                        .flatMap(movie ->
                                reactiveRedisTemplate.opsForValue().set(AppConstants.MOVIE_KEY + movie.id(), movie)
                        .thenReturn(movie)))
                .map(movieMapper::toResponse);
    }
```